modules/vba_documents/lib/vba_documents/multipart_parser.rb

```diff
@@ -81,7 +62,20 @@ def self.create_file_from_base64(infile)
 
       Rails.logger.info("#{self} finished writing Base64-decoded file")
 
-      parse(decoded_file)
+      decoded_file.path
+    end
+
+    def self.validate_virus_free(file_path)
+      temp_path = Common::FileHelpers.generate_clamav_temp_file(file_path)
+      result = Common::VirusScan.scan(temp_path)
+      Common::FileHelpers.delete_file_if_exists(temp_path)
+
+      # Common::VirusScan result will return true or false
+      unless result # unless safe
+        raise VBADocuments::UploadError.new(code: 'DOC101', detail: 'Virus detected in submission')
```

Comment:
This error is handled by the calling class (usually `VBADocuments::UploadProcessor`), which sets the `status` to "error" on the `UploadSubmission` record and also saves the `code` and `detail` to the record.